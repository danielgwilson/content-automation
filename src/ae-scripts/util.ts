/// <reference types="types-for-adobe/aftereffects/2018"/>

function getComp(compName: string) {
  let comp: CompItem | undefined;
  for (let i = 1; i <= app.project.numItems; i++) {
    if (
      app.project.item(i) instanceof CompItem &&
      app.project.item(i).name === compName
    ) {
      comp = app.project.item(i) as CompItem;
      break;
    }
  }
  if (!comp) throw new Error(`Failed to find comp named ${compName}`);
  return comp;
}

function importFootage(filePath: string) {
  const file = new File(filePath);

  if (!file.exists) throw new Error(`File not found: ${filePath}`);

  const importOptions = new ImportOptions(file);
  //importOptions.importAs = ImportAsType.COMP; // you can do stuff like this at this point for PSDs
  return app.project.importFile(importOptions) as AVItem;
}

function addLayer(
  item: AVItem,
  comp: CompItem,
  { name }: { name?: string } = {}
) {
  const layer = comp.layers.add(item);
  layer.name = name ?? layer.name;
  return layer;
}

function duplicateLayer(
  layer: string | number,
  comp: CompItem,
  newName: string
) {
  const target =
    typeof layer === "string" ? comp.layer(layer) : comp.layer(layer);
  const dupe = target.duplicate();
  dupe.name = newName;
  return dupe;
}

function updateTextLayer(layer: string | number, text: string, comp: CompItem) {
  const textLayer = (typeof layer === "string"
    ? comp.layer(layer)
    : comp.layer(layer)) as TextLayer;
  return textLayer.text.sourceText.setValue(new TextDocument(text));
}

function getFootageItem(itemName: string) {
  let item: FootageItem | undefined;
  for (let i = 1; i <= app.project.numItems; i++) {
    if (
      app.project.item(i) instanceof FootageItem &&
      app.project.item(i).name === itemName
    ) {
      item = app.project.item(i) as FootageItem;
      break;
    }
  }
  if (!item) throw new Error(`Failed to find item named ${itemName}`);
  return item;
}
