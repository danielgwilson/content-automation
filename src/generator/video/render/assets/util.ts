export function getSrcForPath(filePath: string) {
  return `file://${filePath}`;
}
