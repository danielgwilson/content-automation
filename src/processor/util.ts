export function getSentences(text: string) {
  return text
    .replace(/(\.+|\:|\!|\?)(\"*|\'*|\)*|}*|]*)(\s|\n|\r|\r\n)/gm, "$1$2|")
    .split("|");
}
