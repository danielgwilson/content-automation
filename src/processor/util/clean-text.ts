import removeMd from "remove-markdown";

export function getCleanText(text: string) {
  let cleanText = text;

  cleanText = removeBadCharacters(cleanText);
  // cleanText = removeProfanity(cleanText);
  cleanText = removeMd(cleanText);

  return cleanText;
}

function removeBadCharacters(text: string) {
  return text.replace(/[\u200B-\u200D\uFEFF]/g, "");
}

function removeProfanity(text: string) {
  const profanity = [
    { word: "fuck", alt: "f-ck" },
    { word: "fucker", alt: "f-cker" },
    { word: "motherfucker", alt: "motherf-cker" },
    { word: "shit", alt: "sh-t" },
    { word: "bullshit", alt: "bullsh-t" },
    { word: "bitch", alt: "b-tch" },
    { word: "cunt", alt: "c-nt" },
    { word: "asshole", alt: "assh-le" },
    { word: "whore", alt: "wh-re" },
    { word: "slut", alt: "sl-t" },
    { word: "cock", alt: "c-ck" },
    { word: "tits", alt: "tits" } // ok
  ];
  let cleanText = text;
  for (let item of profanity) {
    cleanText = cleanText.replace(new RegExp(item.word, "gi"), item.alt);
  }
  return cleanText;
}
