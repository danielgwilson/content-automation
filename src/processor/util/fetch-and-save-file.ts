import fs from "fs";
import path from "path";
import request from "request";

export async function fetchAndSaveFile(
  uri: string,
  { fileName, outputDir }: { fileName: string; outputDir: string }
): Promise<{ fileName: string; filePath: string }> {
  return new Promise(resolve => {
    const filePath = path.join(outputDir, fileName);
    request.head(uri, function(err, res, body) {
      request(uri)
        .pipe(fs.createWriteStream(filePath))
        .on("close", () => {
          return resolve({ fileName, filePath });
        });
    });
  });
}
