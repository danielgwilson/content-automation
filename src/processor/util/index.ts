export { getSentences } from "./get-sentences";
export { trimComments } from "./trim-comments";
export { fetchAndSaveFile } from "./fetch-and-save-file";
