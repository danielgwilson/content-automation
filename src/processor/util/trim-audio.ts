import { IPostSection } from "../../types";
import { getAudioLengthForSections } from "../sections";

export function trimAudio(
  sections: IPostSection[],
  options: { maxAudioLength?: number } = {}
) {
  const { maxAudioLength } = options;
  if (!maxAudioLength) return sections;

  const trimmedSections: IPostSection[] = [];
  let totalAudioLength = 0;
  for (let section of sections) {
    const sectionAudioLength =
      section.length + getAudioLengthForSections(section.children);
    if (totalAudioLength + sectionAudioLength >= maxAudioLength) {
      break;
    }
    trimmedSections.push(section);
    totalAudioLength += sectionAudioLength * 1000;
  }

  return trimmedSections;
}
