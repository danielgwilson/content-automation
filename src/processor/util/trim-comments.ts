import { IPost, IPostComment, IProcessedPostOptions } from "../../types";

export function trimnComments() {}

export function trimComments(
  post: IPost,
  options: { maxRepliesPerComment?: number; maxCommentDepth?: number } = {}
): IPost {
  const { comments, ...postPart } = post;
  const trimmedComments = comments.map(comment => {
    const { replies, ...commentPart } = comment;
    const trimmedReplies = trimReplies(replies, options);
    return { ...commentPart, replies: trimmedReplies } as IPostComment;
  });

  return { ...postPart, comments: trimmedComments };
}

export function trimReplies(
  replies: IPostComment[],
  options: { maxRepliesPerComment?: number; maxCommentDepth?: number } = {}
): IPostComment[] {
  const { maxRepliesPerComment, maxCommentDepth } = options;

  if (maxCommentDepth === undefined || maxCommentDepth > 1) {
    const trimmedReplies = maxRepliesPerComment
      ? replies.slice(0, maxRepliesPerComment)
      : replies;

    return trimmedReplies.map(reply => {
      const { replies, ...replyPart } = reply;

      return {
        ...replyPart,
        replies: trimReplies(replies, {
          maxRepliesPerComment,
          maxCommentDepth: maxCommentDepth ? maxCommentDepth - 1 : undefined
        })
      };
    });
  }

  return [];
}
