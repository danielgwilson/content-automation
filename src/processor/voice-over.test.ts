import config from "config";
import VoiceOverClient from "./voice-over";
import { createContext } from "../util";

describe("voice-over", () => {
  it("Fetches voice over audio files", async () => {
    const context = createContext();
    const GOOGLE_APPLICATION_CREDENTIALS: string = config.get(
      "GOOGLE_APPLICATION_CREDENTIALS"
    );
    const client = new VoiceOverClient(context, {
      GOOGLE_APPLICATION_CREDENTIALS
    });

    const voRequest = {
      text: "This is some test voice over text.",
      fileName: "test-voice-over.mp3",
      subDir: "/test/"
    };
    const audio = await client.fetchVoiceOver(voRequest);

    expect(audio).toMatchSnapshot();
  });
});
