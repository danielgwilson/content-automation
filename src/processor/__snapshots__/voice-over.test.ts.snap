// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`voice-over Fetches voice over audio files 1`] = `
Object {
  "audioConfig": Object {
    "audioEncoding": "MP3",
    "speakingRate": 1,
  },
  "fileName": "test-voice-over.mp3",
  "filePath": "temp/test/test-voice-over.mp3",
  "length": 2.544,
  "voice": Object {
    "languageCode": "en-US",
    "name": "en-US-Wavenet-B",
    "ssmlGender": "MALE",
  },
}
`;
